pliers/extractors/models.py

```diff
@@ -84,3 +84,65 @@ def _extract(self, stim):
             values.append(float(extraction[1]))
 
         return ExtractorResult([values], stim, self, features=features)
+
+
+def _resize_image(image, shape):
+    import numpy as np
+    from PIL import Image
+    return np.array(
+        Image.fromarray(image).resize(shape, resample=Image.BILINEAR))
+
+
+class TensorFlowKerasInceptionV3Extractor(ImageExtractor):
+
+    ''' Labels objects in images using a pretrained Inception V3 architecture
+     implemented in TensorFlow / Keras.
+
+    Args:
+        weights (str): URL to download pre-trained weights. If None (default),
+            uses the pre-trained Inception V3 model (dated 2017-03-10) used in
+            Keras Applications.
+        num_predictions (int): Number of top predicted labels to retain for
+            each image.
+     '''
+
+    _log_attributes = ('weights', 'num_predictions')
+    VERSION = '1.0'
+
+    def __init__(self, weights=None, num_predictions=5):
+        super(TensorFlowKerasInceptionV3Extractor, self).__init__()
+        if weights is None:
+            weights = 'imagenet'
+        self.weights = weights
+        self.num_predictions = num_predictions
+
+        import tensorflow as tf
+        # Instantiating the model also downloads the weights to a cache.
+        self.model = tf.keras.applications.inception_v3.InceptionV3(
+            weights=self.weights)
+
+    def _extract(self, stim):
+        import tensorflow as tf
+
+        required_shape = (299, 299, 3)
+        x = stim.data
+        if x.ndim != 3:
+            raise ValueError("Stim must have rank 3.")
+        if x.shape != required_shape:
+            x = _resize_image(x, required_shape[:-1])
+        x = tf.keras.applications.inception_v3.preprocess_input(x)
+        x = x[None]  # Add batch dimension.
+        preds = self.model.predict(x, batch_size=1)
```

Comment:
Not sure you want to go to the trouble, but in principle, there's a [`BatchTransformerMixin`](https://github.com/tyarkoni/pliers/blob/b9b843c3e383cea2f5d774d471b5499c4038f7d4/pliers/transformers/base.py#L217) that allows you to handle multiple `Stim` objects within `_extract` rather than naively looping. Not sure if it would make a meaningful difference in efficiency in this case.

Reply:
We could do this, but there won't be a meaningful improvement in performance. In my own work, I've found that predicting on single samples in a loop is just as fast as predicting on batches. I don't know why that would be though...

Reply:
That seems odd, just curious what size batches were you testing?

Reply:
The batches were small. This was for a 3D U-Net that worked on inputs of (128, 128, 128, 1). I found that predicting on single samples was as fast as predicting on batches of 4 (I couldn't go any higher due to memory constraints).